src/pynwb/file.py

```diff
@@ -56,6 +56,13 @@ class Subject(NWBContainer):
     @docval({'name': 'age', 'type': str,
              'doc': ('The age of the subject. The ISO 8601 Duration format is recommended, e.g., "P90D" for '
                      '90 days old.'), 'default': None},
+            {
+                "name": "age__reference",
+                "type": str,
+                "doc": "Age is with reference to this event. Can be 'birth' or 'gestational'. If reference is "
+                       "omitted, 'birth' is implied.",
+                "default": "birth",
```

Comment:
For existing files where ``age.reference`` is missing, the user would, I believe, on read see that ``age.reference`` is set to ``birth`` instead of ``None``. I'm not sure if automatically setting this value to ``birth`` when it is missing in a file is desired behavior? I think it may be more intuitive to have ``age.reference`` appear as None if it is missing in the file while still having a default value of `birth` when a user creates a new file. To achieve this, we could either: 1) update the object mapper to set the constructor argument to None if it is missing (or issue a warning that it will be set to ``birth``), or 2) have the default value in ``__init__`` depend on ``self._in_construct_mode``, i.e., ``None if self._in_construct_mode else 'birth'.

Reply:
default already has the behavior you are looking for:

```python
from datetime import datetime
from dateutil.tz import tzlocal, tzutc

from pynwb.file import NWBFile, Subject

from pynwb import NWBHDF5IO

subject = Subject(
    age='P90D',
    #age__reference="birth",
    subject_id='RAT123',
)
nwbfile = NWBFile(
    'a test session description for a test NWBFile',
    'FILE123',
    datetime(2017, 5, 1, 12, tzinfo=tzlocal()),
    session_id='test1',
    subject=subject,
)

with NWBHDF5IO("test_age_ref.nwb", "w") as io:
    io.write(nwbfile)
    
with NWBHDF5IO("test_age_ref.nwb", "r") as io:
    nwb = io.read()
    print(nwb.subject.age__reference)
```
```python
None
```

Reply:
Interesting, I didn't know the ObjectMapper was by default setting it to None if missing (I suspected it would probably just leave it blank and leave the default specified for __init__). This makes sense. Thanks for the demonstration! That's good to know.

Reply:
@oruebel actually I think you were right about this. Now when I read the data back I get "birth" even when it's `None` in the file.

```python
from datetime import datetime
from dateutil.tz import tzlocal, tzutc

from pynwb.file import NWBFile, Subject

from pynwb import NWBHDF5IO

subject = Subject(
    age='P90D',
    age__reference=None,
    subject_id='RAT123',
)
nwbfile = NWBFile(
    'a test session description for a test NWBFile',
    'FILE123',
    datetime(2017, 5, 1, 12, tzinfo=tzlocal()),
    session_id='test1',
    subject=subject,
)

with NWBHDF5IO("test_age_ref.nwb", "w") as io:
    io.write(nwbfile)
    
with NWBHDF5IO("test_age_ref.nwb", "r") as io:
    nwb = io.read()
    print(nwb.subject.age__reference)
```

Reply:
Could you show me how to update the mapper to handle this properly?